Assignment 2.Rmd

```diff
@@ -95,6 +130,45 @@ Now practice with data from our class. Please create a **person-network** with t
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+rm(list=ls())
+library(readr)
+library(tidyr)
+library(dplyr)
+library(igraph)
+a1 <- read.csv("hudk4050-classes.csv",header = TRUE,na.strings = "")
+
+#remove space from class name
+a2<-data.frame(apply(a1,2,function(x) gsub('\\s+','',x)))
+
+#create a name list consisting first name and last name
+a2<-unite(a2,"Name",c("First.Name","Last.Name"),sep="")
+
+#creating the person-class matrix
+a3<-gather(a2,"Class","Class_name", -1, na.rm=TRUE)
+a4<-unique(a3$Class_name)
+a3$count <- 1
+a5<-pivot_wider(a3,id_cols ="Name",names_from ="Class_name",values_from ="count")
+a5[is.na(a5)]<-0
+m1<-as.matrix(a5[,-1])
+rownames(m1)<-a5$Name
+
+#creating the person-person matrix
+m2<-m1%*%t(m1)
+?plot.igraph
+
+#plot the matrix
+g<-graph_from_adjacency_matrix(m2, mode = c("undirected"), weighted = NULL,
+  diag = FALSE)
+plot(g, layout=layout.fruchterman.reingold)
+#generate metrics betweeness centrality and dregree
+betweenness(g, directed = FALSE, weights = NULL, nobigint = TRUE, normalized = FALSE)
+
+```
+The biggest betweeness of cenrality and degree is the student XUDIAN ZHANG
```

Comment:
Good job! Try to interpret the centrality measures within the context of this dataset.